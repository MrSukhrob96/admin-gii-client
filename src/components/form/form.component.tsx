import { hoc } from '@utils/hoc';
import { FormProvider } from 'react-hook-form';
import { useFormProps } from './form.props';
import { FormContainer } from './form.style';

/**
 * <Form />
 */
export const Form = hoc(useFormProps, ({ form, onSubmit, onError }) => (
  <FormProvider {...form}>
    <FormContainer onSubmit={form.handleSubmit(onSubmit, onError)}>
    
    </FormContainer>
  </FormProvider>
));
