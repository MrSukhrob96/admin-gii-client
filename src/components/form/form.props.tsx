import { useCustomForm } from "@hooks";

/**
 * <Form /> props
 */
export type FormProps = {
  form: ReturnType<typeof useCustomForm>;
  onSubmit: (values: unknown) => void;
  onError?: (errors: unknown) => void;
}

export const useFormProps = (_?: FormProps) => ({});

