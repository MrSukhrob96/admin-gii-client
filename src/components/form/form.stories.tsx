import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Form } from './form.component';
import '@styles/index.css';

export default {
  title: 'Common/Form',
  component: Form,
} as ComponentMeta<typeof Form>

const Template: ComponentStory<typeof Form> = (args) => <Form {...args} />;

export const FormInitial = Template.bind({});
FormInitial.args = {
}