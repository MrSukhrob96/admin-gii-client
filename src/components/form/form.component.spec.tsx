import { customRender } from '@test-utils';
import { renderHook } from '@testing-library/react-hooks';
import { useForm } from 'react-hook-form';
import { Form } from './form.component';

describe('Form', () => {
  test('Should render Form component', async () => {
    const { result } = renderHook(() => useForm({
      defaultValues: { test: '' }
    }));

    const onSubmit = jest.fn();

    const { asFragment } = customRender(
      <Form form={result.current} onSubmit={onSubmit} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});