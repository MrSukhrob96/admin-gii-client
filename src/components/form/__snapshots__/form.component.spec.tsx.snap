// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should render Form component 1`] = `
<DocumentFragment>
  <form
    class=""
  />
</DocumentFragment>
`;
